suite-native/storage/src/createAsyncMigrate.ts

```diff
@@ -0,0 +1,70 @@
+import { DEFAULT_VERSION, PersistedState } from 'redux-persist';
+
+export type UnknownPersistedState = unknown & PersistedState;
+
+export type MigrationsManifest = {
+    [key: string]: (
+        state: UnknownPersistedState,
+    ) => UnknownPersistedState | Promise<UnknownPersistedState> | undefined;
```

Comment:
nit: redundant `undefined`, it is already part of the `PersistedState`. But feel free to keep it here if it was intention to make it more explicit.
```suggestion
    ) => UnknownPersistedState | Promise<UnknownPersistedState>;
```

Reply:
I would keep it for the explicitness 🙏